### ISSUE TITLE
Default safe mode per saved connection

### ISSUE THREAD

**Issue by eproxus** [TablePlus/TablePlus#3312]
It would be nice to be able to set the safe mode level in saved connections, so that it is always applied when connecting instead of the global level.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
   ```
   PostgreSQL 16.2
   ```

3. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
   ```
   556
   ```

4. The steps to reproduce this issue:

   1. Configure a default _Safe Mode_ level in _Settings_ (e.g. _Silent Mode_)
   2. Have the desire to set saved production connections **only** to another level (e.g. _Alert Mode 2_)

Previously discussed in #2669.

**Comment by huyphams** [comment]
Have you tried this @eproxus ?
![Screenshot 2024-05-29 at 7 04 08 PM](https://github.com/TablePlus/TablePlus/assets/3654622/c272d9fb-6c7a-49d3-8b9a-c203fa5900a6)

**Comment by eproxus** [comment]
@huyphams Yes, I just realized toggling that switch permanently saved the mode for that connection. I was looking for a setting in the connection edit dialog. Perhaps it can be showed there (or in the open dialog)?

It would be nice if this was at least documented that it works this way (I did check the documentation before but didn't find anything related to safe mode per connection being persisted).
